lib/ukreloc/reloc.c

```diff
@@ -79,18 +78,8 @@ void __used do_ukreloc(__paddr_t r_paddr, __vaddr_t r_vaddr)
 	if (r_vaddr == 0)
 		r_vaddr = (__vaddr_t)rt_baddr;
 
-	/* Since we may have been placed at a random physical address, adjust
-	 * the initial memory region descriptors added through mkbootinfo.py
-	 * since they contain the link-time addresses, relative to rt_baddr.
-	 * Do this before anything else, since `lt_baddr`'s relocation has
-	 * no been resolved yet and contains the link time address.
-	 */
-	ukplat_memregion_foreach(&mrdp, UKPLAT_MEMRT_KERNEL, 0, 0) {
-		mrdp->pbase -= (__paddr_t)lt_baddr;
-		mrdp->pbase += r_paddr;
-		mrdp->vbase -= (__vaddr_t)lt_baddr;
-		mrdp->vbase += r_vaddr;
-	}
+	/* Relocate UKPLAT_MEMRT_KERNEL memory region descriptors */
+	do_ukreloc_kmrds(r_paddr, r_vaddr);
```

Comment:
I will create a stub for it just like for `do_ukreloc`

Reply:
I think the right way is to make it conditional, similarly to calls from other libraries. As discussed, in the future the `ur_` macros will become unconditionally available and the library won't be enabled by default, but that doesn't mean that all other definitions from this library will become unconditionally available too.

Reply:
I do not like having `#ifdef`'s that could have been easily hidden away from the client code. I like having something similar to what linux does: if the CONFIG for this function is not enabled, then use a stub instead. If you place `#ifdef`'s throughout client code it's going to end up being an unreadable, unmaintainable mess. My opinion is that this kind of situations should be transparently handled by the server of the API's.

Reply:
> Linux is not compiled into individual libraries.
Fair, but the build process is largely the same. The burden of `#ifdef`-ing should not fall on the client code. IMO it would be better if it would be handled by the build system.

> In any case, using #ifdef is the way currently done,
And do you not agree with me that it looks like a mess? It's enough to look at `ukboot/boot.c`.

> any proposal to change this scheme shoud be discussed in a broader scope rather being applied ad-hoc.
This library is going to always stay implicitly enabled. When that does not happen I think we should put the burden of `#ifdef`-ing on the build system, e.g. if the library is not enabled, include a dummy header with stubs from `include/uk`.

> And do you not agree with me that it looks like a mess
If you agree with me in this regard, why not begin with this library.

Reply:
> This library is going to stay implicitly enabled for a long time

Actually not, the plan is to address this in the next release.

> If you agree with me in this regard, why not begin with this library.

Because it is not just my call, and it needs to be agreed by the rest of the maintainers.

Reply:
> I will create a stub for it just like for `do_ukreloc`

Actually I was referring to calls to `do_ukreloc` and `do_ukreloc_mrds`, but accidentally placed my comment in the definition instead.

The same comment applies to `do_ukreloc`.

Reply:
I know, I still do not agree.